### PR TITLE
Rifqi/mycite treatment descriptives

### PR DESCRIPTION
**Pull request by RifqiAlkhatib**
descriptives saved into csv. Stitched prescriptions also saved into csv if need to use it again

### CODE REVIEW

File: src/modules/myciteTreatmentDescriptives/myciteTreatmentDescriptives.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    # Fetch all prescriptions for cohort
+    cohort_drugs = nb.get_drug(cohort_ids)
+    cohort_drugs[['start_date', 'end_date']] = cohort_drugs[['start_date', 'end_date']]\
+        .apply(pd.to_datetime)
+
+    # Filter prescriptions in drug_list
+    cohort_filtered_drugs = cohort_drugs[cohort_drugs['drug'].isin(drug_list)]
+    cohort_filtered_drugs['duration_days'] = (
+        cohort_filtered_drugs['end_date'] - cohort_filtered_drugs['start_date']
+    ).dt.days + 1
```
Comment (soonnnanh) [inline src/modules/myciteTreatmentDescriptives/myciteTreatmentDescriptives.py:30]: Why do you plus one here?

File: src/modules/myciteTreatmentDescriptives/myciteTreatmentDescriptives.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+    # Duration on (unstitched) treatment & drug level
+    drug_duration = cohort_filtered_drugs.groupby('drug')['duration_days'].describe()
+
+    # Duration on patient level (unstitched)
+    patient_duration = cohort_filtered_drugs.groupby('person_id')[['duration_days']]\
+        .sum().describe()
+
```
Comment (soonnnanh) [inline src/modules/myciteTreatmentDescriptives/myciteTreatmentDescriptives.py:38]: If we do a groupby on the non-stitched treatment data, and get the duration we will be using the sum of the duration of the treatments even if multiple prescriptions overlap. E.g. If we have 1 prescription from 1/Jan to 30/Jun, and another prescription from 1/Feb to 30/July, assuming each month is 30 days, the total prescription duration for the patient will be 180 + 180 = 360 days. Make this clear either in the function documentation or somewhere

File: src/modules/myciteTreatmentDescriptives/myciteTreatmentDescriptives.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    current_treatment = {}
+    
+    # Iterate through prescriptions (already sorted by start_date)
+    for index, row in df.iterrows():
+        # First treatment
+        if index == 0:
+            current_treatment['person_id'] = row['person_id']
+            current_treatment['start_date'] = row['start_date']
+            current_treatment['end_date'] = row['end_date']
+            current_treatment['treatment_duration_days'] = (row['end_date'] - row['start_date']).days + 1
```
Comment (soonnnanh) [inline src/modules/myciteTreatmentDescriptives/myciteTreatmentDescriptives.py:83]: Remove the + 1

File: src/modules/myciteTreatmentDescriptives/myciteTreatmentDescriptives.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+        else:
+            # Overlapping treatment
+            if row['start_date'] <= (current_treatment['end_date'] + timedelta(days=overlap_threshold)):
+                current_treatment['prescription_overlaps'] += 1
+                # Shift end date if overlapping treatment ends later
+                if row['end_date'] > current_treatment['end_date']:
+                    current_treatment['end_date'] = row['end_date']
+                    current_treatment['treatment_duration_days'] = (
+                        current_treatment['end_date'] - current_treatment['start_date']
+                    ).days + 1
```
Comment (soonnnanh) [inline src/modules/myciteTreatmentDescriptives/myciteTreatmentDescriptives.py:95]: Remove the + 1

File: src/modules/myciteTreatmentDescriptives/myciteTreatmentDescriptives.py [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+            else:
+                df_out = df_out.append(current_treatment, ignore_index=True)
+                current_treatment['start_date'] = row['start_date']
+                current_treatment['end_date'] = row['end_date']
+                current_treatment['treatment_duration_days'] = (row['end_date'] - row['start_date']).days + 1
+                current_treatment['prescription_overlaps'] = 0
+
+    # Append last stitched treatment row    
+    df_out = df_out.append(current_treatment, ignore_index=True)
+    df_out['gap_days'] = (df_out['start_date'] - df_out['end_date'].shift(1)).dt.days
```
Comment (soonnnanh) [inline src/modules/myciteTreatmentDescriptives/myciteTreatmentDescriptives.py:107]: Dont need to add 1 to the day

**Review by soonnnanh — CHANGES_REQUESTED** [review]
Why are the jupyter notebooks pushed into the repo? Besides that and the addition of 1 day that I see in the duration of treatment prescription lines, well done code. 

Note: Please document your code with descriptions of the code and your arguments in the main mycite repo

**Comment by RifqiAlkhatib** [comment]
@soonnnanh Removed notebooks and added documentation so that its clear. +1 days to remain as discussed, should be ok to merge now

**Review by soonnnanh — APPROVED** [review]

**Pull request merged**